plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php

```diff
@@ -183,43 +184,51 @@ protected function get_route_response( \WP_REST_Request $request ) {
 	 * @throws RouteException When a required additional field is missing.
 	 */
 	public function validate_required_additional_fields( \WP_REST_Request $request ) {
-		$contact_fields           = $this->additional_fields_controller->get_fields_for_location( 'contact' );
-		$order_fields             = $this->additional_fields_controller->get_fields_for_location( 'order' );
-		$order_and_contact_fields = array_merge( $contact_fields, $order_fields );
-
-		if ( ! empty( $order_and_contact_fields ) ) {
-			foreach ( $order_and_contact_fields as $field_key => $order_and_contact_field ) {
-				if ( $order_and_contact_field['required'] && ! isset( $request['additional_fields'][ $field_key ] ) ) {
-					throw new RouteException(
-						'woocommerce_rest_checkout_missing_required_field',
-						/* translators: %s: is the field label */
-						esc_html( sprintf( __( 'There was a problem with the provided additional fields: %s is required', 'woocommerce' ), $order_and_contact_field['label'] ) ),
-						400
-					);
+		$document_object = null;
+
+		if ( Features::is_enabled( 'experimental-blocks' ) ) {
+			$document_object = new DocumentObject(
+				[
+					'customer' => [
+						'billing_address'  => $request['billing_address'],
+						'shipping_address' => $request['shipping_address'],
+					],
+					'checkout' => [
+						'payment_method'    => $request['payment_method'],
+						'create_account'    => $request['create_account'],
+						'customer_note'     => $request['customer_note'],
+						'additional_fields' => $request['additional_fields'],
+					],
+				]
+			);
+		}
+
+		if ( WC()->cart->needs_shipping() ) {
+			$shipping_fields = $this->additional_fields_controller->get_fields_for_group( 'shipping' );
```

Comment:
This does the same thing but less confusing no?
```suggestion
			$shipping_fields = $this->additional_fields_controller->get_fields_for_location( 'address' );
```

Reply:
I guess either works since the former calls the later.